dkist/io/asdf/tests/test_models.py

```diff
@@ -37,6 +41,63 @@ def test_roundtrip_vct():
     assert u.allclose(new_ivct(*world, 5*u.pix), pixel[:2], atol=0.01*u.pix)
 
 
+def test_roundtrip_vct_2d():
+    varying_matrix_lt = [rotation_matrix(a)[:2, :2] for a in np.linspace(0, 90, 15)] * u.arcsec
+    varying_matrix_lt = varying_matrix_lt.reshape((5, 3, 2, 2))
+
+    vct = VaryingCelestialTransform2D(crpix=(5, 5) * u.pix,
+                                      cdelt=(1, 1) * u.arcsec/u.pix,
+                                      crval_table=(0, 0) * u.arcsec,
+                                      pc_table=varying_matrix_lt,
+                                      lon_pole=180 * u.deg)
+    new_vct = roundtrip_object(vct)
+    assert isinstance(new_vct, VaryingCelestialTransform2D)
+    new_ivct = roundtrip_object(vct.inverse)
+    assert isinstance(new_ivct, InverseVaryingCelestialTransform2D)
+
+    assert u.allclose(u.Quantity(new_vct.crpix), (5, 5) * u.pix)
+    assert u.allclose(u.Quantity(new_ivct.crpix), (5, 5) * u.pix)
+
+    assert u.allclose(u.Quantity(new_vct.pc_table), varying_matrix_lt)
+    assert u.allclose(u.Quantity(new_ivct.pc_table), varying_matrix_lt)
+
+    pixel = (0*u.pix, 0*u.pix, 4*u.pix, 2*u.pix)
+    world = new_vct(*pixel)
+    assert u.allclose(world, (359.99861111, 0.00138889)*u.deg)
+
+    assert u.allclose(new_ivct(*world, 4*u.pix, 2*u.pix), pixel[:2], atol=0.01*u.pix)
+
+
+def test_roundtrip_vct_slit():
+    varying_matrix_lt = [rotation_matrix(a)[:2, :2]
+                         for a in np.linspace(0, 90, 10)] * u.arcsec
+
+    vct = VaryingCelestialTransformSlit(crpix=(5, 5) * u.pix,
```

Comment:
The 2D refers to the shape of the lookup table.

The non-slit models take one or two extra pixel inputs which are used to index the lookup table.

The slit variants of the models use one less extra pixel input, as the pointing varies along the second pixel dimension (step in the raster).

There is no mention of time here because all of these models only work in pixel space. Even for the inverse (world to pixel) transform the time coordinates are transformed to pixel and then passed to the inverse.

Hopefully that makes sense!

Reply:
Oh and also, these models only output spatial axes, the wave and temporal parts are done by other models.